### PR TITLE
Add com.pikatorrent.PikaTorrent

### PR DESCRIPTION
**Pull request by G-Ray**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. 
The `.desktop` file & `metainfo.xml` files as been added to the upstream project: https://github.com/G-Ray/pikatorrent/commit/e7457351670e130fe824e657ce4174c37dc28085

I tried to compile the project from sources, but `flatpak-node-generator` does not seem to work with a lockfile v3 with workspaces (monorepo). I opened https://github.com/flatpak/flatpak-builder-tools/issues/366

As a workaround, I converted the lockfile to v2, but fetching a cached dependecny from github failed. Issue seems similar to https://github.com/flatpak/flatpak-builder-tools/issues/182, but I need to investigate further.

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: com.github.G_Ray.PikaTorrent.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+app-id: com.github.G_Ray.PikaTorrent
```
Comment (hfiguiere) [inline com.github.G_Ray.PikaTorrent.yml:1]: it should be `io.github`

File: com.github.G_Ray.PikaTorrent.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  - --socket=pulseaudio
+  - --share=network
+cleanup:
+modules:
+  - name: pikatorrent
+    buildsystem: simple
+    build-commands:
+      # Copy
+      - mv pikatorrent/* /app/
+      - rmdir pikatorrent
```
Comment (hfiguiere) [inline com.github.G_Ray.PikaTorrent.yml:23]:
```suggestion
```

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: that's disappointing

File: com.github.G_Ray.PikaTorrent.yml (hunk 1 above)
Comment (G-Ray) [reply com.github.G_Ray.PikaTorrent.yml]: Thanks, actually I control pikatorrent.com so I updated the app id to `com.pikatorrent.PikaTorrent`

File: flathub.json (hunk 3 above)
Comment (G-Ray) [reply flathub.json:2]: Yes, ARM builds will be possible once https://github.com/G-Ray/transmission-native/issues/20 is resolved :rocket:

Comment (hfiguiere) [reply flathub.json:2]: you could build it here instead.

**Comment by hfiguiere** [comment]
bot build com.pikatorrent.PikaTorrent

**Comment by hfiguiere** [comment]
webp isn't a supported screenshot format. That's the build error.

**Comment by G-Ray** [comment]
> webp isn't a supported screenshot format. That's the build error.

Thanks, I updates the screenshots links to png files.

File: flathub.json (hunk 3 above)
Comment (G-Ray) [reply flathub.json:2]: I would like, I hope I will be able to resolve the issues I described in the PR comment in the near future, to build from sources :crossed_fingers:

**Pull request closed without merging**